### ISSUE TITLE
cut words one by one

### ISSUE THREAD

**Issue by CoolDarran** [rkcosmos/deepcut#23]
hi @rkcosmos 

when I ran the example, I got the a different result.

```python
>>> import deepcut
Using TensorFlow backend.
2017-10-30 18:32:12.384678: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
2017-10-30 18:32:12.384717: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
2017-10-30 18:32:12.384726: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
2017-10-30 18:32:12.384732: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
2017-10-30 18:32:12.384739: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
2017-10-30 18:32:12.679174: I tensorflow/core/common_runtime/gpu/gpu_device.cc:955] Found device 0 with properties:
name: GeForce GTX 1080 Ti
major: 6 minor: 1 memoryClockRate (GHz) 1.6325
pciBusID 0000:02:00.0
Total memory: 10.91GiB
Free memory: 10.50GiB
2017-10-30 18:32:12.910366: W tensorflow/stream_executor/cuda/cuda_driver.cc:523] A non-primary context 0x5583d11dd3f0 exists before initializing the StreamExecutor. We haven't verified StreamExecutor works with that.
2017-10-30 18:32:12.911434: I tensorflow/core/common_runtime/gpu/gpu_device.cc:955] Found device 1 with properties:
name: GeForce GTX 1080 Ti
major: 6 minor: 1 memoryClockRate (GHz) 1.6325
pciBusID 0000:03:00.0
Total memory: 10.91GiB
Free memory: 10.75GiB
2017-10-30 18:32:12.912357: I tensorflow/core/common_runtime/gpu/gpu_device.cc:976] DMA: 0 1
2017-10-30 18:32:12.912376: I tensorflow/core/common_runtime/gpu/gpu_device.cc:986] 0:   Y Y
2017-10-30 18:32:12.912415: I tensorflow/core/common_runtime/gpu/gpu_device.cc:986] 1:   Y Y
2017-10-30 18:32:12.912429: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1045] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:02:00.0)
2017-10-30 18:32:12.912439: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1045] Creating TensorFlow device (/gpu:1) -> (device: 1, name: GeForce GTX 1080 Ti, pci bus id: 0000:03:00.0)
>>> print(" ".join(deepcut.tokenize(u'ตัดคำได้ดีมาก')))
ต ั ด ค ำ ไ ด ้ ด ี ม า ก
>>> print(deepcut.tokenize('ตัดคำได้ดีมาก'))
['\xe0', '\xb8', '\x95', '\xe0', '\xb8', '\xb1', '\xe0', '\xb8', '\x94', '\xe0', '\xb8', '\x84', '\xe0', '\xb8', '\xb3', '\xe0', '\xb9', '\x84', '\xe0', '\xb8', '\x94', '\xe0', '\xb9', '\x89', '\xe0', '\xb8', '\x94', '\xe0', '\xb8', '\xb5', '\xe0', '\xb8', '\xa1', '\xe0', '\xb8', '\xb2', '\xe0', '\xb8', '\x81']
```

I install deepcut by pip. Am I missing some configurations?

**Comment by tienlemanh** [comment]
I have the same problem. @rkcosmos @titipata

**Comment by titipata** [comment]
@tienlemanh, can you give more information about this (Python version/ tensorflow version)?  It seems like this is the problem with string type.

**Comment by tienlemanh** [comment]
@titipata Python 2.7.9 tensorflow 1.2.1

```
>> pip show tensorflow
Name: tensorflow
Version: 1.2.1
Summary: TensorFlow helps the tensors flow
Home-page: http://tensorflow.org/
Author: Google Inc.
Author-email: opensource@google.com
License: Apache 2.0
Location: /usr/local/lib/python2.7/dist-packages
Requires: werkzeug, markdown, six, html5lib, protobuf, backports.weakref, wheel, numpy, mock, bleach
```

**Comment by tienlemanh** [comment]
@titipata 
```
>>> print ' '.join(deepcut.tokenize('ตัดคำได้ดีมาก'))
� � � � � � � � � � � � � � � � � � � � � � � � � � � � � � � � � � � � � � �
>>>
>>> print ' '.join(deepcut.tokenize(u'ตัดคำได้ดีมาก'))
deepcut/utils.py:68: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  if char in CHARS:
ต ั ด ค ำ ไ ด ้ ด ี ม า ก
```

**Comment by titipata** [comment]
@tienlemanh, @CoolDarran It seems to be problem with string encoding in Python 2. Basically, Python 2 cannot handle unicode well. When it reads dictionary or Thai text, it turns to non-unicode which messed up with the mapping character to index. 

I still haven't found a way to fix it. If anyone can help, please keep me posted here!

**Comment by tienlemanh** [comment]
@titipata I already fixed it. It works now. I changed code from str text to unicode text in everywhere. Should I create a pull request, then you can see?

**Comment by titipata** [comment]
@tienlemanh, that's cool! Please send the PR!

**Comment by titipata** [comment]
Closed by PR #24.
